<a href="https://colab.research.google.com/github/amandapdb/PrevixTor/blob/main/Aprofundamento_Estatistico_Amanda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
from google.colab import files
uploaded = files.upload()

Saving processed_data.csv to processed_data.csv


In [3]:
df = pd.read_csv('processed_data.csv')

In [4]:
df.head()

,Air temperature [K],Process temperature [K],Rotational speed [rpm],Torque [Nm],Tool wear [min],Machine failure,TWF,HDF,PWF,OSF,RNF,Type_H,Type_L,Type_M
0,-0.936508,-0.935111,0.248279,0.271766,-1.697534,0,0,0,0,0,0,False,False,True
1,-0.886191,-0.867785,-0.862295,0.674931,-1.649807,0,0,0,0,0,0,False,True,False
2,-0.936508,-1.002438,-0.163332,1.032021,-1.617989,0,0,0,0,0,0,False,True,False
3,-0.886191,-0.935111,-0.668139,-0.108362,-1.586170,0,0,0,0,0,0,False,True,False
4,-0.886191,-0.867785,-0.862295,-0.050767,-1.554352,0,0,0,0,0,0,False,True,False


# Identificação do Problema

Durante a análise exploratória, foi observada uma forte correlação positiva entre a temperatura do ar e a temperatura do processo.

O objetivo desta análise é verificar se essa relação é estatisticamente significativa ou se pode ter ocorrido ao acaso.

## Teste de Normalidade

In [5]:
from scipy.stats import shapiro

stat_air, p_air = shapiro(df['Air temperature [K]'])
stat_process, p_process = shapiro(df['Process temperature [K]'])

print("Air temperature p-value:", p_air)
print("Process temperature p-value:", p_process)

Air temperature p-value: 3.1649695996516864e-33
Process temperature p-value: 3.169764221949781e-22


/usr/local/lib/python3.12/dist-packages/scipy/stats/_axis_nan_policy.py:579: UserWarning: scipy.stats.shapiro: For N > 5000, computed p-value may not be accurate. Current N is 9256.
  res = hypotest_fun_out(*samples, **kwds)


Air temperature p-value: 3.16e-33
Process temperature p-value: 3.16e-22
Ambos são p < 0,05, ou seja, os dados não seguem uma distribuição normal. Por isso vamos usar testes não paramétricos, aqui no caso, o Spearmen

## Formulação de Hipóteses

H0 (Hipótese Nula): Não existe correlação entre a temperatura do ar e a temperatura do processo.

H1 (Hipótese Alternativa): Existe correlação entre a temperatura do ar e a temperatura do processo.

##Execução do Teste Estatístico

In [6]:
from scipy.stats import spearmanr

corr, p_value = spearmanr(df['Air temperature [K]'], df['Process temperature [K]'])

print("Correlação:", corr)
print("p-value:", p_value)

Correlação: 0.8658311089830775
p-value: 0.0


Foi utilizado o teste de correlação de Spearman, devido à ausência de normalidade nos dados.

O resultado apresentou uma correlação de 0.86, indicando uma forte relação positiva entre a temperatura do ar e a temperatura do processo.

O tamanho do efeito foi avaliado com base no coeficiente de correlação de Spearman, que apresentou valor de 0.86.

Esse valor indica um efeito forte, demonstrando que a relação entre as variáveis analisadas é significativa não apenas estatisticamente, mas também na prática.

O p-valor encontrado foi inferior a 0,05 (p ≈ 0.0), permitindo rejeitar a hipótese nula.


## Conclusão Científica

Com base nos resultados obtidos, conclui-se que a correlação observada entre a temperatura do ar e a temperatura do processo é estatisticamente significativa, não sendo fruto do acaso. Podendo assim, descartar a hipótese nula.

Além disso, o valor da correlação indica um efeito forte, demonstrando relevância prática para o problema analisado.

No contexto do projeto, essa relação sugere que a temperatura do ar é um fator importante para prever o comportamento da temperatura do processo, sendo boas features, podendo ser utilizada como variável relevante em modelos futuros de Machine Learning. é preciso ter atenção porque as vezes a correlação muito alta de duas variáveis pode acarretar na duplicidade de dados, porém em contrapartida, aumentam a capacidade preditiva do modelo.

Foi utilizado o teste de correlação de Spearman, pois o objetivo da análise foi verificar a relação entre duas variáveis contínuas.

Os testes como Teste-t, ANOVA e Mann-Whitney não foram utilizados, pois são aplicados em comparações entre grupos, o que não corresponde ao objetivo desta análise.